src/internal/kopia/data_collection.go

```diff
@@ -49,19 +64,49 @@ func (kdc kopiaDataCollection) Fetch(
 	ctx context.Context,
 	name string,
 ) (data.Stream, error) {
-	if kdc.snapshotRoot == nil {
-		return nil, clues.New("no snapshot root")
+	ctx = clues.Add(ctx, "item_name", clues.Hide(name))
+
+	if kdc.dir == nil {
+		return nil, clues.New("no snapshot directory")
+	}
+
+	if len(name) == 0 {
+		return nil, clues.Wrap(errNoRestorePath, "getting item stream").WithClues(ctx)
+	}
+
+	e, err := kdc.dir.Child(ctx, encodeAsPath(name))
+	if err != nil {
+		if isErrEntryNotFound(err) {
+			err = clues.Stack(data.ErrNotFound, err)
+		}
+
+		return nil, clues.Wrap(err, "getting item").WithClues(ctx)
+	}
+
+	f, ok := e.(fs.File)
+	if !ok {
+		return nil, clues.New("object is not a file").WithClues(ctx)
+	}
+
+	size := f.Size() - int64(versionSize)
```

Comment:
Does this need a `size` < 0 error check?

Reply:
hmmm probably a good check to have. I'll add a log message so we know it happened. Ideally it shouldn't though because this should be the version that's injected by the backup logic